actionpack/lib/action_dispatch/journey/route.rb

```diff
@@ -140,7 +135,7 @@ def required_defaults
       end
 
       def glob?
-        ast.glob?
+        path.ast.glob?
```

Comment:
I'll take a close look later today, but I think `route.path.spec.any?(Nodes::Star)` would work in place of `route.glob?`. That's basically what the implementation of `route.glob?` used to before https://github.com/rails/rails/pull/39935.

Reply:
`route.path.spec.any?(Nodes::Star)` does the job, thank you.

I appreciate that you will take a closer look. Please let me know the results (even if you don't have a chance to take a look, please let me know, so that I can release a fix 😁).

Reply:
OK, I've refreshed my memory on this.

`Route#glob?` used to do `path.spec.any?(Nodes::Star)`, visiting every node to see if any of them were stars. https://github.com/rails/rails/pull/39935 changed `Route#glob?` to get the value from a new `Journey::Ast` object instead. `Journey::Ast` improves the performance of adding routes by visiting the nodes upfront and caching various values along the way so we don't have to visit them again.

Once Rails is done adding the route, it doesn't need the `Journey::Ast` anymore. To avoid retaining memory for objects that we won't use, we [clear out all the `Journey::Ast` objects as part of eager loading](https://github.com/rails/rails/pull/39935/files#diff-b5197cabb760000bb11b71067d918d9b5c246c42a9b47563ee0e7b221add7bfcR31).

I was aware that clearing out the `Journey::Ast` would break some methods, but these methods were all private Rails methods that Rails doesn't need again after adding the route. I think it's possible to get the `glob?` method working again by caching it's value, but I'll defer to folks on the core team about whether we want that. It would add an extra instance variable to every route, which might not be desirable to support a private method that Rails doesn't use after eager loading.

`route.path.spec` is also technically private, but various applications are using it and it's less likely to change (I actually did attempt to change it, but brought back the original behavior in this PR because folks were relying on it). I see that airbrake is already using `route.path.spec`, so using it instead of `route.glob?` might be good anyway to limit airbrake's dependence on private Rails methods.

Reply:
Thanks for your elaborative answer. I appreciate it. I will use `route.path.spec.any?(Nodes::Star)` instead.
You may want to consider exposing routes in a public API since this is useful for Application Performance Monitoring (APM). Airbrake monitors performance of an app's routes, and therefore we need to know the routes.